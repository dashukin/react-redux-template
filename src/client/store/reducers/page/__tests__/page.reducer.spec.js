import forOwn from 'lodash/forOwn';

import pageReducer, { defaultState } from '../page.reducer';
import pageConstans from '../page.constants';

describe('pages.reducer', () => {
  describe('page default state', () => {
    it('should match snapshot', () => {
      expect(defaultState).toMatchSnapshot();
    });
  });

  describe('page transformation', () => {
    it('should return default state', () => {
      const page = pageReducer(undefined, {
        type: undefined,
      });

      expect(page).toBe(defaultState);
    });

    it('should return page listed in page constants', () => {
      forOwn(pageConstans, (value, key) => {
        const action = {
          type: key,
        };
        const page = pageReducer(undefined, action);

        expect(page).toEqual(value);
      });
    });
  });
});
