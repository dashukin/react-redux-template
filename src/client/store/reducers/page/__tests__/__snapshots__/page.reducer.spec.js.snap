// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages.reducer page default state should match snapshot 1`] = `"HOME"`;
